main/src/ca/uwaterloo/flix/language/ast/Symbol.scala

```diff
@@ -413,7 +414,14 @@ object Symbol {
   /**
     * Restrictable Enum Symbol.
     */
-  final class RestrictableEnumSym(val namespace: List[String], val name: String, val loc: SourceLocation) extends Symbol {
+  final class RestrictableEnumSym(val namespace: List[String], val name: String, cases: List[Name.Ident], val loc: SourceLocation) extends Symbol {
+
+    // NB: it is critical that this be either a lazy val or a def, since otherwise `this` is not fully instantiated
+    /**
+      * The universe of cases associated with this restrictable enum.
+      */
+    lazy val universe: SortedSet[Symbol.RestrictableCaseSym] = cases.map(Symbol.mkRestrictableCaseSym(this, _)).to(SortedSet)
```

Comment:
I my prefer a def.

Reply:
What is a lazy val, if not a cached def?

I guess this wouldn't exactly be a bottleneck.